meetings/2023-04-06.md

```diff
@@ -0,0 +1,96 @@
+* JM: Is cURL described in the current spec?
+* eP: metnions authn issue
+* eP: It is not presented to the user - used at runtime but not exposed. In Some systems, webhook for example, it can be copied. here it is in the subscription response but not to human interaction.
+* JM: As I understand it, the URL is response. That's the complex URL?
+* eP: Yes.
+* JM: The developer for app receiving it, they wouldn't be able to setup an endpoint my/app/parameteruuid and feed into library. Not difficult but something to think about.
+* eP: If you compare to authn/z in place, with tokens, setting some access controls on the endpoint.. this might be easier ot have a lightweight receiver and have all this authn/z system. There is a tradeoff.
+* JM: The solution to authn/z can be implemented in a stateless manner whereas cURLs need state maintained on the NR side. The NR needs to remember the id of the cURL during the subscription phase. Whereas if you use a token the authn is inherit in the token without needing to maintain state. Things from the implementation level. From a spec, cURL has fewer open questions.
+* eP: re state of the URL that needs to maintained, but if you talk about in the previous version of the Webhooks, if you have the sender in the response, you need to keep the sender, and add it to the ACL on the receiver. SubscriptionClient gets the sender, so that NR knows about it, and then when the notification comes, NR checks if it comes from NS.
+* JM: Depending on the biz logic, might need to maintain the state for what you're subscribed to. But that is not necessarily the UC. Can imagine UC where NR... as I receive it, I will then be able to validate this notification came from where it claims to come from, and do something with that data based on that.
+* JM: If you want to have a stateless app.. it is possible with the new authn scheme.. that token is cryptographically signed. But under cURL, I have a random uuid that came in.. and depending on my memory to say that "ah yes! this is something i subscribed to before". It is tiny things. Not hard for a client to maintain state but this is the trade off with cURL and entire new authn scheme.
+* JM: Is preventing spam within the purview of notification spec? Or just want a secure way to send it. Should spam be accepted or be concerned about DDOS attacks? Someone directs a bunch of notifications to one server in which case .. even under cURL under DDOS attacks.
+* eP: We can track separate issue for spam because of nuances. If we can have a system that gets deployed and lots of spam going on we'd be hurt by that if we recommend. People will hate it.
+* JM: I guess that was one of hte considerations in the webhook specification. Traditionally with the webhook system someone will manually sets it up. But because this automatic setup we might need a way for new authn scheme.. own the domain for pointing the webhook to.
```

Comment:
```suggestion
* eP: We can track separate issue for spam because of nuances. If we can have a system that gets deployed and lots of spam going on, we'd be hurt by that if we recommend. People will hate it.
* JM: I guess that was one of the considerations in the webhook specification. Traditionally with the webhook system, someone will manually sets it up. But because this is automatic setup, we might need a way for new authn scheme... own the domain for pointing the webhook to.
```